src/librustc_codegen_ironox/constant.rs

```diff
@@ -0,0 +1,23 @@
+use type_::Type;
+
+/// An unsigned constant.
+///
+/// It must be possible to represent this constant using 128 bits.
```

Comment:
I think this comment means that `value` must fit within 128 bits (rather than the struct itself, which is clearly bigger than 128 bits)? Is there a particular reason for this? I wouldn't mind knowing what the consequences of ignoring this are (even if we just say "or the results are undefined").

Reply:
I guess there has to be an upper bound somewhere, as Rust doesn't do bignum out of the box.

Reply:
That's right, all integer constants must fit within 128 bits. In fact, this restriction is imposed by the `ConstMethods` trait the backend has to implement. All of the `ConstMethods` that produce integer constants take an argument which represents the value of the constant. This argument is always at most 128 bits wide.